src/hooks/useBreakpoint.tsx

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from "react";
+import { useDebouncedCallback } from "use-debounce";
+import { BreakpointKey, Breakpoints } from "../Css";
+
+/**
+ * A React hook to return a record of responsive breakpoints that updates on resize.
+ *
+ * @example
+ * const { breakpoints } = useBreakpoint();
+ * if(breakpoints.mdAndDown) {...do something cool}
+ */
+
+type BreakpointsType = Partial<Record<BreakpointKey, boolean>>;
+
+export function useBreakpoint(): { breakpoints: BreakpointsType } {
+  const [breakpoints, setBreakpoints] = useState(matchMediaBreakpoints());
+
+  useEffect(() => {
+    const handleResize = useDebouncedCallback(
+      // function
+      () => {
+        setBreakpoints(matchMediaBreakpoints());
```

Comment:
**suggestion:** Per my comment:
> Can we ensure that the hook only triggers a re-render of the calling components when there is a change to the matched breakpoints?
This is where we'd want to ensure something changed before we call `setBreakpoints`. Otherwise we're saying a state changed, even if it didn't really change.

Reply:
added check before setting so we reduce the unneeded re-renders etc.